src/orca/services/synapse/ops.py

```diff
@@ -57,3 +61,64 @@ def monitor_evaluation_queue(self, evaluation_id: str) -> bool:
         )
         submissions_num = sum(1 for submission in received_submissions)
         return submissions_num > 0
+
+    def trigger_indexing(self, synapse_view: str) -> None:
+        """
+        Trigger indexing in Synapse.
+
+        Arguments:
+            syn: A Synapse object.
+            synapse_view: The Synapse ID of the view.
+        """
+        self.client.tableQuery(f"select * from {synapse_view} limit 1")
+
+    def get_submissions_with_status(
```

Comment:
Does this return an empty list when there are no submissions with the matching status?

Reply:
Actually it wasn't in the original screenshots - I've just updated them in the description box for record-keeping. Here is a screenshot answering your question. If none of the submissions have the status fed to `get_submissions_with_status` the list returned is empty and basically nothing happens. I'm going to add some error handling to catch typos/incorrect status types, but for status types [that do exist](https://rest-docs.synapse.org/rest/org/sagebionetworks/evaluation/model/SubmissionStatusEnum.html) but none of the submissions happen to be labeled with it, do we want some error-handling to catch this and break the Airflow run? Or do we want Airflow to quietly stop?

<img width="903" alt="image" src="https://github.com/Sage-Bionetworks-Workflows/py-orca/assets/32107699/5886bfba-e9a0-443b-a511-db25a40eceb8">

Reply:
My plan was for an empty list returned to lead to the `stop` [task](https://github.com/Sage-Bionetworks-Workflows/orca-recipes/blob/4e6a92892ff5381c5d979b622f9f1de52ba64243/dags/challenge-submission-dag.py#L51) from a branching task. As for status error handling, the Synapse API already does a decent job handling this with an enum:
```
challengeutils change-status 9742660 ABC

...

> synapseclient.core.exceptions.SynapseHTTPError: 400 Client Error:
> No enum constant org.sagebionetworks.evaluation.model.SubmissionStatusEnum.ABC
```

Reply:
Sounds good, then I will leave the logic as it is for statuses that do exist in [here](https://rest-docs.synapse.org/rest/org/sagebionetworks/evaluation/model/SubmissionStatusEnum.html) but not assigned to any submission in the designated view.

I agree the error-handling for `update_submission_status` looks fine (wraps `challengeutils`). I've added error-handling for `get_submissions_with_status` since that was returning an empty list even if the status was just not a valid status, so figure it would be more helpful in tracking down the problem if you ever have to. See my changes here https://github.com/Sage-Bionetworks-Workflows/py-orca/pull/37/commits/085837376f6cf7145440d37dc2a5f65488518a72

Reply:
Oh, I see what you are saying. Cool, I'll take a closer look when you are ready for the next round of review

Reply:
Yep I'm ready